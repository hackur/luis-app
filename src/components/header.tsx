import * as React from 'react';
import { Header } from 'semantic-ui-react';

export const TopHeader = () => <Header icon="user" content="Application Header" />;