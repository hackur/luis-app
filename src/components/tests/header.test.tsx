import * as React from 'react';
import { Header } from '../header';
import { mount } from 'enzyme';

// this is extended 'expectatio' library from 'https://github.com/mjackson/expect'
// we are not using 'jests' one as jest cannot run in browser

describe('Header', () => {

  story('Default View', 'Some **description**', () => {
    const component = <Header />;

    it('Renders correctly', function () {
      expect(mount(component)).toMatchSnapshot();
    });

    return component;
  })

  describe('With Decorator', () => {
    decorator((story) => (
      <div style={{ padding: '15px', background: 'red' }}>
        {story()}
      </div>
    ));

    story('Decorated View', 'Some **other description**', () => {
      const component = <Header />;

      it('Renders correctly', function () {
        expect(mount(component)).toMatchSnapshot();
      });

      return component;
    })

  });
});