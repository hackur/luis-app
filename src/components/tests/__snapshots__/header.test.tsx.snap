// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Default View Renders correctly 1`] = `
<Component>
  <h1>
    Header 123
  </h1>
</Component>
`;

exports[`Header With Decorator Decorated View Renders correctly 1`] = `
<Component>
  <Header
    content="Application Header"
    icon="user"
  >
    <div
      className="ui header"
    >
      <Icon
        as="i"
        name="user"
      >
        <i
          aria-hidden="true"
          className="user icon"
        />
      </Icon>
      <HeaderContent>
        <div
          className="content"
        >
          Application Header
        </div>
      </HeaderContent>
    </div>
  </Header>
</Component>
`;
